src/core/gltf2exporter/copyingexportpass_p.h

```diff
@@ -0,0 +1,122 @@
+/*
+    copyingexportpass_p.h
+
+    This file is part of Kuesa.
+
+    Copyright (C) 2019 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+    Author: Jean-Michaël Celerier <jean-michael.celerier@kdab.com>
+
+    Licensees holding valid proprietary KDAB Kuesa licenses may use this file in
+    accordance with the Kuesa Enterprise License Agreement provided with the Software in the
+    LICENSE.KUESA.ENTERPRISE file.
+
+    Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef KUESA_GLTF2EXPORTER_COPYINGEXPORTPASS_P_H
+#define KUESA_GLTF2EXPORTER_COPYINGEXPORTPASS_P_H
+
+//
+//  NOTICE
+//  ------
+//
+// We mean it: this file is not part of the public API and could be
+// modified without notice
+//
+
+#include "gltf2exporter_p.h"
+#include "gltf2keys_p.h"
+#include "gltf2uri_p.h"
+
+#include <QJsonArray>
+#include <QJsonObject>
+
+QT_BEGIN_NAMESPACE
+namespace Kuesa {
+namespace {
+
+class CopyingExportPass
+{
+public:
+    CopyingExportPass(
+            QDir source,
+            QDir destination)
+        : m_basePath{ source }
+        , m_destination(destination)
+    {
+    }
+
+    void addGeneratedFiles(const QStringList &lst)
+    {
+        m_generated << lst;
+    }
+
+    const QStringList &errors() const { return m_errors; }
+
+    void copyURIs(QJsonObject &root, const QLatin1String &key)
+    {
+        using namespace GLTF2Import;
+        if (m_basePath == m_destination)
+            return;
+
+        auto array_it = root.find(key);
+        if (array_it == root.end())
+            return;
+
+        const auto &array = array_it->toArray();
+        for (const auto &val : array) {
+            const auto &buffer = val.toObject();
+            auto uri_it = buffer.find(GLTF2Import::KEY_URI);
+            if (uri_it == buffer.end())
+                continue;
+
+            const auto uri = uri_it->toString();
+            if (Uri::kind(uri) != Uri::Kind::Path)
+                continue;
+
+            if (m_generated.contains(uri))
+                continue;
+
+            QFile srcFile(Uri::localFile(uri_it->toString(), m_basePath));
+            const QFileInfo destFile(m_destination, uri_it->toString());
+
```

Comment:
Shouldn't we overwrite if the file already exists ?

Reply:
done